### ISSUE TITLE
Warnings with hardcoded local paths when dwarf-with-dsym is enabled

### ISSUE THREAD

**Issue by debug45** [facebook/facebook-ios-sdk#2031]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/main/CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Xcode version

13.2.1

### Facebook iOS SDK version

13.0.0

### Dependency Manager

SPM

### SDK Framework

Core

### Goals

I just build my project

### Expected results

Absence of such warnings

### Actual results

```
/Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/1PB9PMN2JHX2N/FBSDKCoreKit-1JNRK9IV71RVR.pcm: No such file or directory
/Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/1PB9PMN2JHX2N/Foundation-3MV0PSJQONL78.pcm: No such file or directory
/Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/1PB9PMN2JHX2N/SwiftShims-DPJP6D9F3FJI.pcm: No such file or directory
```

**Comment by jawwad** [comment]
@debug45 - Wow, really weird. Thanks for reporting. I'll take a look.

**Comment by jawwad** [comment]
We don't have any hardcoded paths, at least with my name within the repo itself so this must be some weird side effect of how we are building. Also, this was the first release built with Xcode 13.0.0 instead of Xcode 12.5.0 so perhaps some build behavior changed. (We build with older versions for some time to provide backward compatibility, since Xcode guarantees forward compatibility but not backward compatibility, i.e. a release built with Xcode 13 may not work with Xcode 12).

**Comment by samodom** [comment]
@debug45 We're having trouble reproducing this issue with a sample app.  Do you have a simple app that you can share showing this behavior?

**Comment by adarhef** [comment]
This is happening to me as well:
```
[16:27:11]: ▸     /Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/2DMG2GR6APLXZ/FBSDKCoreKit-27WT04A4HUT75.pcm: No such file or directory
[1711](https://github.com/stuff-that-works/ios/runs/5661934030?check_suite_focus=true#step:8:1711)
[16:27:11]: ▸     /Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/2DMG2GR6APLXZ/Foundation-15A17VFA0X01X.pcm: No such file or directory
[1712](https://github.com/stuff-that-works/ios/runs/5661934030?check_suite_focus=true#step:8:1712)
[16:27:11]: ▸     /Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/2DMG2GR6APLXZ/SwiftShims-1KP8S06RFNK11.pcm: No such file or directory
[1713](https://github.com/stuff-that-works/ios/runs/5661934030?check_suite_focus=true#step:8:1713)
[16:27:11]: ▸     /Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/2DMG2GR6APLXZ/FBSDKLoginKit-2WA2FFE43LSEU.pcm: No such file or directory
[1714](https://github.com/stuff-that-works/ios/runs/5661934030?check_suite_focus=true#step:8:1714)
[16:27:11]: ▸     /Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/2DMG2GR6APLXZ/CoreGraphics-1IOXVD9XVXJZV.pcm: No such file or directory
[1715](https://github.com/stuff-that-works/ios/runs/5661934030?check_suite_focus=true#step:8:1715)
[16:27:11]: ▸     /Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/2DMG2GR6APLXZ/FBSDKCoreKit-2T5UQ6M47YZ9E.pcm: No such file or directory
[1716](https://github.com/stuff-that-works/ios/runs/5661934030?check_suite_focus=true#step:8:1716)
[16:27:11]: ▸     /Users/jawwad/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/2DMG2GR6APLXZ/UIKit-3EUHNXEBEPAJN.pcm: No such file or directory
[1717](https://github.com/stuff-that-works/ios/runs/5661934030?check_suite_focus=true#step:8:1717)
```

**Comment by jawwad** [comment]
@debug45, @adarhef - Could you update to the iOS SDK 13.1.0 and see if you still encounter the issue? (Note: You may see /Users/samodom/... instead of /Users/jawwad... for 13.1.0 if the issue still persists)

**Comment by adarhef** [comment]
> @debug45, @adarhef - Could you update to the iOS SDK 13.1.0 and see if you still encounter the issue? (Note: You may see /Users/samodom/... instead of /Users/jawwad... for 13.1.0 if the issue still persists)

> @debug45, @adarhef - Could you update to the iOS SDK 13.1.0 and see if you still encounter the issue? (Note: You may see /Users/samodom/... instead of /Users/jawwad... for 13.1.0 if the issue still persists)

[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/Foundation-2Y409IXV0UXR5.pcm: No such file or directory
[3039](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3039)
[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/FBAEMKit-1ZCAMUBFTI9AU.pcm: No such file or directory
[3040](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3040)
[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/FBSDKCoreKit-21U55KRQO5SZ.pcm: No such file or directory
[3041](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3041)
[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/FBSDKCoreKit_Basics-1E7N8SJ6E21AI.pcm: No such file or directory
[3044](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3044)
[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/UIKit-2P4UZ2DCUH474.pcm: No such file or directory
[3045](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3045)
[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/FBSDKLoginKit-3RQ3NV7BHL9OD.pcm: No such file or directory
[3046](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3046)
[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/CoreGraphics-299CWW130W0CP.pcm: No such file or directory
[3047](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3047)
[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/FBSDKCoreKit-26XS7D95NBQSV.pcm: No such file or directory
[3048](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3048)
[08:43:24]: ▸     /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/8LUBLU1SEAK0/SwiftShims-RJ9EA3NQYL57.pcm: No such file or directory
[3049](https://github.com/stuff-that-works/ios/runs/5672018760?check_suite_focus=true#step:6:3049)

Indeed, now it's samodom

**Comment by jawwad** [comment]
@debug45, @adarhef - Could you see if this is happening in a brand new empty project for you? You can create a new app or use the [FacebookTest.zip](https://github.com/facebook/facebook-ios-sdk/files/8339911/FacebookTest.zip) sample project in https://github.com/facebook/facebook-ios-sdk/issues/2047#issuecomment-1077389896.

**Comment by adarhef** [comment]
> 

I just tried running FacebookTest.zip and didn't get any warnings, so I guess it's somehow related to my setup. Any pointers maybe?

**Comment by joshuapoq** [comment]
Enabling `DWARF with dSYM file` for debug produces these warnings on the FirebaseTest.zip @jawwad @adarhef. We have this enabled based on Firebase install guide for its automatic symbol uploads. I used to disable this but gave up fighting Firebase on that one. I assume these warnings would also show on Release config builds.

**Comment by jawwad** [comment]
@joshuapoq - Thanks for figuring this out! I can confirm that changing from "dwarf" to "dwarf-with-dsym" reproduces this issue. So changing ""dwarf-with-dsym" to "dwarf" for the **Debug Information Format** setting will get rid of these warnings for the Debug config. Now the next question to figure out is why these hardcoded paths are there in the first place.

**Comment by wooseongkim-29cm** [comment]
@jawwad Is there any update on this? We had same issue on our project. Hope this will be fixed on next update 🙏 

```
/Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/ ~
...
```

**Comment by weakfl** [comment]
@jawwad any update?

**Comment by username0x0a** [comment]
This warning seems even more relevant for production builds:


```
Linking a static library that was built with -gmodules, but the module cache was not found.  Redistributable static libraries should never be built with module debugging enabled.  The debug experience will be degraded due to incomplete debug information.
```

Hoping for a quick fix. 👍

**Comment by joshuapoq** [comment]
Just for my two cents. I'm not convinced this is an issue specifically on Facebook's side. I'd think this is more likely an issue Xcode 13 or SPM has introduced. I've seen similar warnings from some of our frameworks but we do have some legacy Objective-C dependencies. Would love to hear what you find from this!

**Comment by username0x0a** [comment]
Same output when archiving for App Store, combined with the message I've already suggested:

```
Linking a static library that was built with -gmodules, but the module cache was not found.  Redistributable static libraries should never be built with module debugging enabled.  The debug experience will be degraded due to incomplete debug information.
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit-DXOV6DW0LNJQ.pcm: No such file or directory while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit-DXOV6DW0LNJQ.pcm
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/Foundation-A3SOD99KJ0S9.pcm: No such file or directory while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/Foundation-A3SOD99KJ0S9.pcm
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit_Basics-HRK1V9LMU6Z4.pcm: No such file or directory while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit_Basics-HRK1V9LMU6Z4.pcm
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/UIKit-2LM3EQU7VVY4O.pcm: No such file or directory while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/UIKit-2LM3EQU7VVY4O.pcm
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/SwiftShims-3A5RGGAXA7QR7.pcm: No such file or directory while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/SwiftShims-3A5RGGAXA7QR7.pcm
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBAEMKit-1V9NEHYJ0RPPR.pcm: No such file or directory while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBAEMKit-1V9NEHYJ0RPPR.pcm
```

**Comment by twprzybysz** [comment]
Same output: 
```
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/1FTMPQGUROQW7/Foundation-3DFYNEBRQSXST.pcm: No such file or directory
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/1FTMPQGUROQW7/FBSDKCoreKit-2A74VT7G965SU.pcm: No such file or directory
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/1FTMPQGUROQW7/FBSDKShareKit-3T4EP89CWOAGB.pcm: No such file or directory
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/1FTMPQGUROQW7/Photos-3A71D5TFDRRPT.pcm: No such file or directory
/Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/1FTMPQGUROQW7/UIKit-1V5UHAPTOD24G.pcm: No such file or directory
(...)
```

**Comment by tb-rendity** [comment]
We're getting these same warnings as others.

Its a case of enabling DWARF-with-dsym to let crashlytics debug builds to effectively provide test data or put up with these somewhat annoying warnings.

Could we see a fix for this soon? We have a set of best practices that this doesn't really fit well with

**Comment by adarhef** [comment]
Still happening in latest 14.1.0

```
/Users/joesusnick/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/35QDORLDX1QGG/FBSDKCoreKit-3D3JA1F3KLL0X.pcm: No such file or directory
```

**Comment by AlexNsbmr** [comment]
Any news about this issue?
Still happening with Xcode 13.4.1 and Facebook SDK 14.1.0

**Comment by tb-rendity** [comment]
@AlexNsbmr Still happening on SDK 14.1.0

**Comment by skware** [comment]
On our side we get the following warnings since we installed the SDK with SPM:
Any update on this ?
Thanks

warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit-DXOV6DW0LNJQ.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit-DXOV6DW0LNJQ.pcm
note: Linking a static library that was built with -gmodules, but the module cache was not found.  Redistributable static libraries should never be built with module debugging enabled.  The debug experience will be degraded due to incomplete debug information.
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/Foundation-A3SOD99KJ0S9.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/Foundation-A3SOD99KJ0S9.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit_Basics-HRK1V9LMU6Z4.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit_Basics-HRK1V9LMU6Z4.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/UIKit-2LM3EQU7VVY4O.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/UIKit-2LM3EQU7VVY4O.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/SwiftShims-3A5RGGAXA7QR7.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/SwiftShims-3A5RGGAXA7QR7.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBAEMKit-1V9NEHYJ0RPPR.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBAEMKit-1V9NEHYJ0RPPR.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/CoreGraphics-1TH4RJJRQVFOC.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/CoreGraphics-1TH4RJJRQVFOC.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit-1PD9RL8CDABK7.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit-1PD9RL8CDABK7.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKLoginKit-181EZAM29R1ID.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKLoginKit-181EZAM29R1ID.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit_Basics-3GWE6P1P29USP.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/FBSDKCoreKit_Basics-3GWE6P1P29USP.pcm
warning: /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/ObjectiveC-1KD62J152BYGO.pcm: No such file or directory
note: while processing /Users/josuebrizuelaxp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/22AY12HINWDP6/ObjectiveC-1KD62J152BYGO.pcm

**Comment by joshuapoq** [comment]
Please can we get an update on this @jawwad at least on what investigation has been done and why this one has been and will be hard to fix?

I'd guess this is caused by some parameter set in the project or passed to build commands within your tooling. From my own experience I'd guess it's either an easy fix or it'd require a tooling rewrite but I'd love to know why this is taking so long to fix.

We have around 30 projects using Facebook and these warnings are a frustration against our code health policies.

**Comment by piercifani** [comment]
I can't believe this is still not fixed... Think about how many teams that can't properly keep a healthy codebase because they integrate with the FB SDK.

**Comment by zvjerka24** [comment]
On my end similar situation but with local path (SDK version 14.0.0)

`warning build: /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/ZJ2DA96MT3UV/FBSDKCoreKit-2Z9E41DRBTW5F.pcm: No such file or directory
warning build: /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/ZJ2DA96MT3UV/FBSDKCoreKit_Basics-1E7N8SJ6E21AI.pcm: No such file or directory
warning build: /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/ZJ2DA96MT3UV/FBAEMKit-1KL8MADBFRQEQ.pcm: No such file or directory
warning build: /Users/samodom/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/ZJ2DA96MT3UV/FBSDKCoreKit_Basics-2N84BNN6R7MIU.pcm: No such file or directory`

**Comment by samodom** [comment]
We made some updates in the v15.0.0 release that we're hoping will fix this issue.  Can anyone currently experiencing this please try that version to see if is solved for them?

**Comment by skware** [comment]
For me upgrading to 15.0.0 seems to remove those warnings. 
Glad to have my warnings pane empty again!
Thanks!

**Comment by joesus** [comment]
Closing for now since this is fixed as of 15.0.0.
